pkg/tenantctl/controller.go

```diff
@@ -184,21 +184,21 @@ func (c *Controller) syncHandler(key string) error {
 
 	tenant := origTenant.DeepCopy()
 
-	c.logPrintln(tenant, "started reconciliation")
-	defer c.logPrintln(&api.Tenant{ObjectMeta: *tenant.ObjectMeta.DeepCopy()}, "finished reconciliation")
+	c.logPrintln(4, tenant, "started reconciliation")
+	defer c.logPrintln(4, &api.Tenant{ObjectMeta: *tenant.ObjectMeta.DeepCopy()}, "finished reconciliation")
```

Comment:
Here we should avoid needless object creations:
```suggestion
	if klog.V(4).enabled() {
		defer c.logPrintln(4, &api.Tenant{ObjectMeta: *tenant.ObjectMeta.DeepCopy()}, "finished reconciliation")
	}
```

Reply:
It's neither resolved or commented.

Reply:
The if statement klog.V(4).enabled() is not required. According to the documentation:
`if glog.V(2).Enabled() { klog.Info("log this") }` == ` klog.V(2).Info("log this")`
So the .V() methods logs only when verbosity is set.

Reply:
@LenaHoinkis You missed my point and this sentence from the docs:
> The second form is shorter but the first is cheaper if logging is off because it does not evaluate its arguments.

Reply:
Is a defer really that expensive in Go that we need to boil our code with additional if statements?

Reply:
> Is a defer really that expensive in Go that we need to boil our code with additional if statements?

It's not about the `defer`, but the object creation with deep copy + its garbage collection afterwards that can be saved here.